channeldb/graph_test.go

```diff
@@ -2846,7 +2863,17 @@ func TestGraphZombieIndex(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unable to create test vertex: %v", err)
 	}
+
+	// Swap the nodes if the second's pubkey is smaller than the first.
+	// Without this, the comparisons at the end will fail probabilistically.
+	if bytes.Compare(node2.PubKeyBytes[:], node1.PubKeyBytes[:]) < 0 {
+		node1, node2 = node2, node1
+	}
+
 	edge, _, _ := createChannelEdge(db, node1, node2)
+	if err := graph.AddChannelEdge(edge); err != nil {
```

Comment:
The check below assumes that the graph doesn't yet know of this edge. So why does adding it to the graph not cause the test to fail?

Reply:
the comment is now incorrect. since the edge is known, it shouldn't be zombie